src/main/scala/net/psforever/actors/session/ChatActor.scala

```diff
@@ -1333,4 +1176,190 @@ class ChatActor(
         ))
     }
   }
+
+  def customCommandMessages(
+                             message: ChatMsg,
+                             session: Session,
+                             chatService: ActorRef[ChatService.Command],
+                             cluster: ActorRef[InterstellarClusterService.Command],
+                             gmCommandAllowed: Boolean
+                           ): Boolean = {
+//    val messageType = message.messageType
+//    val recipient = message.recipient
+    val contents = message.contents
+    if (contents.startsWith("!")) {
+      if (contents.startsWith("!whitetext ") && gmCommandAllowed) {
+        chatService ! ChatService.Message(
+          session,
+          ChatMsg(UNK_227, true, "", contents.replace("!whitetext ", ""), None),
+          ChatChannel.Default()
+        )
+        true
+
+      } else if (contents.startsWith("!loc ")) {
+        val continent = session.zone
+        val player = session.player
+        val loc =
+          s"zone=${continent.id} pos=${player.Position.x},${player.Position.y},${player.Position.z}; ori=${player.Orientation.x},${player.Orientation.y},${player.Orientation.z}"
+        log.info(loc)
+        sessionActor ! SessionActor.SendResponse(message.copy(contents = loc))
+        true
+
+      } else if (contents.startsWith("!list")) {
+        val zone = contents.split(" ").lift(1) match {
+          case None =>
+            Some(session.zone)
+          case Some(id) =>
+            Zones.zones.find(_.id == id)
+        }
+        zone match {
+          case Some(inZone) =>
+            sessionActor ! SessionActor.SendResponse(
+              ChatMsg(
+                CMT_GMOPEN,
+                message.wideContents,
+                "Server",
+                "\\#8Name (Faction) [ID] at PosX PosY PosZ",
+                message.note
+              )
+            )
+            (inZone.LivePlayers ++ inZone.Corpses)
+              .filter(_.CharId != session.player.CharId)
+              .sortBy(p => (p.Name, !p.isAlive))
+              .foreach(player => {
+                val color = if (!player.isAlive) "\\#7" else ""
+                sessionActor ! SessionActor.SendResponse(
+                  ChatMsg(
+                    CMT_GMOPEN,
+                    message.wideContents,
+                    "Server",
+                    s"$color${player.Name} (${player.Faction}) [${player.CharId}] at ${player.Position.x.toInt} ${player.Position.y.toInt} ${player.Position.z.toInt}",
+                    message.note
+                  )
+                )
+              })
+          case None =>
+            sessionActor ! SessionActor.SendResponse(
+              ChatMsg(
+                CMT_GMOPEN,
+                message.wideContents,
+                "Server",
+                "Invalid zone ID",
+                message.note
+              )
+            )
+        }
+        true
+
+      } else if (contents.startsWith("!ntu") && gmCommandAllowed) {
```

Comment:
instead of `else if` it might be cleaner to use pattern matching

`case a if a.startsWith("...") =>`

Reply:
Also having the content of the operation be in a function might make it cleaner too:

```scala
command match {
    case .... => doThis()
    case .... => doThat()
}
````